### ISSUE TITLE
Cannot move file_handle through make<> by rvalue reference

### ISSUE THREAD

**Issue by AliveDevil** [microsoft/cppwinrt#778]
Consider following IDL:
```
[default_interface]
runtimeclass Sample { }
```
Header:
```
namespace implementation
{
    struct Sample : SampleT<Sample>
    {
        Sample(file_handle& handle);

    private:
        file_handle handle;
    };
}
```
Implementation:
```
namespace implementation
{
	Sample::Sample(file_handle& handle) : handle(handle) {
	}
}
```
With following usage:
```
file_handle file{ CreateFile(basePath.data(), WRITE_DAC, 7, nullptr, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL | FILE_FLAG_BACKUP_SEMANTICS, nullptr) };
if (!file) {
	return nullptr;
}
return make<Sample>(file);
```
There is no way of triggering the rvalue reference copy constructor of `winrt::handle_type<winrt::file_handle_traits>` as it is always deleted.
Following usages always generate compile issues:
* `Sample::Sample(file_handle& handle) : handle(handle)` and `Sample::Sample(file_handle&& handle) : handle(handle)`
> C2280	"winrt::handle_type<winrt::file_handle_traits>::handle_type(const winrt::handle_type<winrt::file_handle_traits> &)" : Es wurde versucht, auf eine gelöschte Funktion zu verweisen
> E1776	Auf "Funktion "winrt::handle_type<T>::handle_type(const winrt::handle_type<winrt::file_handle_traits> &) [mit T=winrt::file_handle_traits]" (implizit deklariert)" kann nicht verwiesen werden (ist eine gelöschte Funktion).
* Instead of copy constructor using the assignment operator (`this->handle = handle`)
> C2280	"winrt::handle_type<winrt::file_handle_traits> &winrt::handle_type<winrt::file_handle_traits>::operator =(const winrt::handle_type<winrt::file_handle_traits> &)" : Es wurde versucht, auf eine gelöschte Funktion zu verweisen
> E1776	Auf "Funktion "winrt::handle_type<T>::operator=(const winrt::handle_type<winrt::file_handle_traits> &) [mit T=winrt::file_handle_traits]" (implizit deklariert)" kann nicht verwiesen werden (ist eine gelöschte Funktion).

Is there any way of moving a `file_handle` through `winrt::make`?
The only work around I found would be to use `handle(handle.detach())` inside the constructor assignment, which is far from ideal.

**Comment by DefaultRyan** [comment]
`winrt::make` uses perfect forwarding to preserve rvalue references for your types. But, in multiple places, your code isn't properly invoking the rvalue (move) constructor

```c++
namespace implementation
{
    struct Sample : SampleT<Sample>
    {
        Sample(file_handle&& handle) // Accept an rvalue reference in your constructor
            : handle(std::move(handle))  // Use handle's rvalue constructor
        {}

    private:
        file_handle handle;
    };
}

make<Sample>(file{ /* stuff */ }); // Provide make<> with an rvalue instead of an lvalue

// or

file_handle file{ /* stuff */ };
make<Sample>(std::move(file));
```
